tests/scripts/generate_test_code.py

```diff
@@ -184,7 +184,7 @@
 END_CASE_REGEX = r'/\*\s*END_CASE\s*\*/'
 
 DEPENDENCY_REGEX = r'depends_on:(?P<dependencies>.*)'
-C_IDENTIFIER_REGEX = r'!?[a-z_][a-z0-9_]*$'
+C_IDENTIFIER_REGEX = r'!?[a-z_][a-z0-9_]*(([<>]=?|==)[0-9]*)?$'
```

Comment:
Please keep `C_IDENTIFIER_REGEX` the way it was and define a new macro for an identifier with a numerical constraint.

Reply:
OK, but it is used only in `validate_dependency()`, and there the numerical constraint is also sent to be validated

Reply:
What I'm suggesting is to remove `validate_dependency`. Let the C preprocessor handle the job.

Reply:
It is always best to catch problems as early as possible. in this case, in the python script, and not during compile time.
If one has a typo causing an invalid dependency, I would expect it to be raised while generating the tests, and not during compile time, or worse, on runtime, because any  character works